airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/arrayformater/ObsoleteArrayFormatter.java

```diff
@@ -0,0 +1,61 @@
+package io.airbyte.integrations.destination.bigquery.formatter.arrayformater;
+
+import static io.airbyte.integrations.destination.bigquery.formatter.DefaultBigQueryDenormalizedRecordFormatter.PROPERTIES_FIELD;
+import static io.airbyte.integrations.destination.bigquery.formatter.DefaultBigQueryDenormalizedRecordFormatter.TYPE_FIELD;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ObsoleteArrayFormatter extends ModernArrayFormatter {
```

Comment:
nit: technically, this is not `obsolete`, because it is still being used, but `legacy`, so let's rename it to `LegacyArrayFormatter`